### PR TITLE
Feature/forgottenpasswordendpoint

### PR DESCRIPTION
**Pull request by AWitcherONS**
### What

Added the backend forgotpasswordconfirmation endpoint

### How to review
ensure that
Unit tests 
Feature tests
Check code for style 

### Who can review

Anyone except me

### CODE REVIEW

File: api/users.go [hunk 1]
```diff
@@ -213,9 +213,24 @@ func (api *API) ChangePasswordHandler(ctx context.Context, w http.ResponseWriter
...
+		validationErrs := changePasswordParams.ValidateForgottenPasswordRequiredRequest(ctx)
+		if len(validationErrs) != 0 {
+			return nil, models.NewErrorResponse(validationErrs, http.StatusBadRequest, nil)
+		}
+		changeForgottenPasswordRequest := changePasswordParams.BuildConfirmForgotPasswordRequest(api.ClientSecret, api.ClientId)
+
+		_, cognitoErr := api.CognitoClient.ConfirmForgotPassword(changeForgottenPasswordRequest)
+
+		if cognitoErr != nil {
+			parsedErr := models.NewCognitoError(ctx, cognitoErr, "ConfirmForgottenPassword request, NEW_PASSWORD_REQUIRED type, from change password endpoint")
```
Comment (mwnicks) [inline api/users.go]: `NEW_PASSWORD_REQUIRED type` only applies to the other change password process, it should be removed from the error context here.

File: features/users.feature [hunk 2]
```diff
@@ -467,13 +467,13 @@ Feature: Users
                 "session": "auth-challenge-session"
             }
             """
-        Then I should receive the following JSON response with status "501":
+        Then I should receive the following JSON response with status "400":
```
Comment (mwnicks) [inline features/users.feature:470]: The scenario description here no longer matches whats being tested.

File: features/users.feature [hunk 3]
```diff
@@ -804,3 +804,317 @@ Feature: Users
             }
         """
         Then the HTTP status code should be "202"
+
+    Scenario: POST /v1/password-reset non ONS email address and checking the response status 202
```
Comment (mwnicks) [inline features/users.feature]: There seems to be duplicate scenarios

**Review by mwnicks — CHANGES_REQUESTED** [review]

File: features/users.feature [hunk 4]
```diff
@@ -1,5 +1,5 @@
 Feature: Users
-
+ƒ
```
Comment (mike-kelly-methods) [inline features/users.feature]: This can be removed!

File: api/users_test.go [hunk 5]
```diff
@@ -5,12 +5,12 @@ import (
 	"context"
 	"encoding/json"
 	"errors"
-	"github.com/aws/aws-sdk-go/aws/awserr"
 	"net/http"
 	"net/http/httptest"
 	"testing"
 
 	"github.com/ONSdigital/dp-identity-api/models"
+	"github.com/aws/aws-sdk-go/aws/awserr"
```
Comment (mike-kelly-methods) [inline api/users_test.go:13]: Is this needed?

File: features/users.feature (hunk 2 above)
Comment (AWitcherONS) [reply features/users.feature:470]: I had to leave one mistake to ensure that you were checking

File: api/users.go (hunk 1 above)
Comment (AWitcherONS) [reply api/users.go]: changed error message to "ConfirmForgottenPassword request from change password endpoint"

File: api/users.go [hunk 6]
```diff
@@ -213,9 +213,24 @@ func (api *API) ChangePasswordHandler(ctx context.Context, w http.ResponseWriter
...
+			// change string
```
Comment (mike-kelly-methods) [inline api/users.go]: Is this comment required? If so, can it be a bit more descriptive?

**Review by mwnicks — DISMISSED** [review]

**Review by mike-kelly-methods — APPROVED** [review]
👍

**Pull request merged**